File: miscellaneous/945. Minimum Increment to Make Array Unique.java
// https://leetcode.com/problems/minimum-increment-to-make-array-unique/
class Solution {
    public int minIncrementForUnique(int[] A) {
        if(A.length <= 1)return 0;
        Arrays.sort(A);
        int currMax = A[0];
        int count = 0;
        for(int i = 1; i < A.length; i++){
            if(A[i] > currMax){
                currMax = A[i];
            }
            else{
                currMax++;
                count += currMax - A[i];
            }
        }
        return count;
    }
}

File: miscellaneous/945. Minimum Increment to Make Array Unique.go
// https://leetcode.com/problems/minimum-increment-to-make-array-unique/description/?envType=daily-question&envId=2024-06-14
func minIncrementForUnique(nums []int) int {
    count := 0
    sort.Ints(nums)

    for i := 1; i < len(nums); i++ {
        if nums[i] <= nums[i - 1] {
            increment := nums[i - 1] - nums[i] + 1
            count += increment

            nums[i] = nums[i - 1] + 1
        }
    }
    return count
}